### ISSUE TITLE
BUG: Getting Started instructions / Docker Compose

### ISSUE THREAD

**Issue by jeffmaher** [CovidShield/server#24]
Saw this project and was hoping to get it running locally using the `docker-compose` configuration. 

Alas, it didn't work for a clean run. Do you have getting started instructions for Docker?

# Steps

1. Clone this repo
1. Run `docker-compose up`

# Expected

Docker to be running `server_key-retrieval` on port `8001`, `server_key-submission` on port `8000`, and a `mysql` instance.

# Result

`server_key-retrieval` doesn't stay up.

Docker processes look like this, where `server_key-retrieval` continually restarts:

```
√>docker ps -a
CONTAINER ID        IMAGE                   COMMAND                   CREATED             STATUS                          PORTS                    NAMES
c331f2dcc069        server_key-retrieval    "/bin/sh -c \"/usr/lo…"   8 minutes ago       Restarting (2) 12 seconds ago                            server_key-retrieval_1
aede2e4cf4c2        server_key-submission   "/bin/sh -c \"/usr/lo…"   8 minutes ago       Up 8 minutes                    0.0.0.0:8000->8000/tcp   server_key-submission_1
5a172ada98c1        mysql:5.7               "docker-entrypoint.s…"    8 minutes ago       Up 8 minutes                    3306/tcp, 33060/tcp      server_mysql_1
```

Here's a log file of a `docker-compose up` command, running for a bit then killing the process after ~20 minutes:

[log.txt](https://github.com/CovidShield/server/files/4672597/log.txt)

**Comment by LouisStAmour** [comment]
Since this ticket is on documentation, I'll share here some documentation I was writing just now in a Slack channel:

It might be obvious to some, it wasn't obvious to me -- `KEY_CLAIM_HOST` referred to by Portal is meant to point to the **key-submission** server so with local/default docker-compose ports, that would be `localhost:8000` in the portal's docker-compose.yml.

**KEY_CLAIM_TOKEN** is the bearer token from a client's perspective, as in https://github.com/CovidShield/server/blob/master/examples/new-key-claim/node.js#L13 or multiple bearer tokens for the **key-submission** server, as in the example at  https://github.com/CovidShield/server/blob/ca905dab42a8a65b8d8432e271a0515426dffbdb/pkg/keyclaim/authenticator.go#L18-L20 - what this means is the portal's KEY_CLAIM_TOKEN doesn't have `=` in it, just one of the parts before the `=`. The server, on the other hand, can have multiple valid tokens passed to it, separated by `:` . In the existing docker-compose.yml files, a region of `1` is defined with the secret key `test` while the same secret key is referenced by portal (`test`) which means the portal is configured for region `1` by default.

To quote the readme:
> A centralized, shared database allows diagnosis keys to be distributed to all provinces and territories across Canada. Meanwhile, per-province/territory deployment and management of the mobile apps and web-based results portals allows each region to operate in accordance with its own guidelines while still taking advantage of the national, shared database.

So the intent then is that each region would have a secret hexadecimal key paired to the region identifier in the format `key=ON:key=QC:...` to be loaded by the key-submission server. Region identifiers are effectively limited to the regex `[0-9A-Za-z_]+` by the Go server app and appear to be treated as strings otherwise.

The other usage for region IDs appears to be retrieving exposure configuration as mentioned at https://github.com/CovidShield/server/blob/0a5eb818ab1ece471129d4c11542c6d318fb2559/README.md#retrieving-exposure-configuration -- however this is currently incomplete/undefined beyond a static response from the key-retrieval server in https://github.com/CovidShield/server/blob/0a5eb818ab1ece471129d4c11542c6d318fb2559/pkg/server/config.go#L25 -- ideally it would use a config file to vary the response by region, I suppose. It's arguable regions might want to independently update the data, but that coordination could be done out-of-band if not self-serve...

**Comment by LouisStAmour** [comment]
There's also some preliminary documentation at https://github.com/CovidShield/server/blob/7530e2189088653f5e5dbf2c1b53e9731e0274b4/config/infrastructure/aws/README.md

Arguably, documentation, or repos, should be centralized such that there's one obvious place to look for documentation on the whole project. Right now reading source code (frantic CTRL+F, I mean) has been the most illuminating way to understand the app. 😃

**Comment by Ginja** [comment]
Partially closed in #28. `docker-compose up` should work now.

We'll get some docs up soon to assist with getting started.

**Comment by jeffmaher** [comment]
Yay! Thanks @Ginja !
